rts/lra/test/arquillian-extension/src/main/java/io/narayana/lra/arquillian/deployment/scenario/ScenarioGeneratorBase.java

```diff
@@ -0,0 +1,237 @@
+    Map<String, String> getExtensionProperties(final String extensionName) {
+
+        ArquillianDescriptor arquillianDescriptor = arquillianDescriptorInstance.get();
+
+        Optional<Map<String, String>> checkExistance = arquillianDescriptor.getExtensions().stream()
+                .filter(x -> x.getExtensionName().equals(extensionName))
+                .map(ExtensionDef::getExtensionProperties).findAny();
+
+        if (!checkExistance.isPresent()) {
+            String message = String.format("%s: there is not extension section with name %s defined in the %s file",
+                    this.getClass().getSimpleName(),
+                    extensionName,
+                    System.getProperty("arquillian.xml", "arquillian.xml"));
+
+            log.warn(message);
+        }
+
+        return checkExistance.orElse(null);
+    }
+
+    /**
+     * This methods checks that the properties in the {@link List} <code>toCheck</code>
+     * are present in the {@link Map} <code>properties</code>
+     * @param properties {@link Map} representing the properties of an extension
+     * @param toCheck {@link List} of properties that need to be defined
+     * @throws RuntimeException if a property is not found
+     */
+    void checkPropertiesExistence(final Map<String, String> properties, final List<String> toCheck) {
+        // if not all properties in toCkeck are defined in properties
+        if (!toCheck.stream().allMatch(properties::containsKey)) {
+            // tell me what property need to be defined
+            toCheck.forEach(x -> {
+                if (!properties.containsKey(x)) {
+                    String message = String.format("%s: the property %s is not defined in the extension",
+                            ScenarioGeneratorBase.class.getSimpleName(),
+                            x);
+
+                    log.warn(message);
+                }
+            });
+        }
+    }
+
+    /**
+     * This method extracts a {@link Method} that represents a deployment method that
+     * should be called to create an {@link org.jboss.shrinkwrap.api.Archive} to deploy
+     * in an Arquillian container
+     * @param properties {@link Map} representing the properties of an extension
+     * @param deploymentMethodPropertyName name of the deployment method property
+     * @return {@link Method}
+     * @throws ClassNotFoundException if the specified class is not valid
+     * @throws NoSuchMethodException if the specified method is not valid
+     * @throws RuntimeException if anything else goes wrong
+     */
+    Method getDeploymentMethodFromConfiguration(final Map<String, String> properties, final String deploymentMethodPropertyName)
+            throws ClassNotFoundException, NoSuchMethodException, RuntimeException{
+
+        String property = properties.get(deploymentMethodPropertyName).trim();
+
+        String className = property.substring(0, property.lastIndexOf('.'));
+        Class<?> clazz = Class.forName(className);
```

Comment:
Class loading and working with specific property names...

Reply:
Although these names are defined in the main class (`LRACoordinatorScenarioGenerator`), I can also document them in the javadoc. I thought that using a configurable approach (reflection + property names and deployment function) would have been easier to follow/understand...I see that (maybe) this is not the case